src/AutoMapper/XpressionMapper/Extensions/MapperExtensions.cs

```diff
@@ -33,8 +34,18 @@ public static TDestDelegate MapExpression<TDestDelegate>(this IMapper mapper, La
             var remappedBody = visitor.Visit(expression.Body);
             if (remappedBody == null)
                 throw new InvalidOperationException(Resource.cantRemapExpression);
+
+            remappedBody = VerifyType(typeDestFunc.GetGenericArguments().Last(), remappedBody);
```

Comment:
Simply replacing VerifyType with ToType should be enough here.

Reply:
Good  point.